cmake/modules/FindSleef.cmake

```diff
@@ -61,10 +61,28 @@ find_library(Sleef_LIBRARY
   PATHS ${PC_Sleef_LIBRARY_DIRS})
 mark_as_advanced(Sleef_LIBRARY)
 
-find_library(SleefDFT_LIBRARY
-  NAMES sleefdft
-  PATHS ${PC_Sleef_LIBRARY_DIRS})
-mark_as_advanced(SleefDFT_LIBRARY)
+foreach(_component ${Sleef_FIND_COMPONENTS})
+  find_library(Sleef_${_component}_LIBRARY
+    NAMES sleef${_component}
+    PATHS ${PC_Sleef_LIBRARY_DIRS})
+  mark_as_advanced(Sleef_${_component}_LIBRARY)
+  set(ADDITIONAL_REQUIRED_VARS ${ADDITIONAL_REQUIRED_VARS} Sleef_${_component}_LIBRARY)
+
+  if(Sleef_${_component}_LIBRARY)
+    set(Sleef_${_component}_FOUND TRUE)
+    if(NOT TARGET Sleef::${_component})
+      add_library(Sleef::${_component} UNKNOWN IMPORTED)
+      set_target_properties(Sleef::${_component}
+        PROPERTIES
+          IMPORTED_LOCATION "${Sleef_${_component}_LIBRARY}"
+          INTERFACE_COMPILE_OPTIONS "${PC_Sleef_CFLAGS_OTHER}"
+          INTERFACE_INCLUDE_DIRECTORIES "${Sleef_INCLUDE_DIR}"
+      )
+    endif()
+  else()
+    set(Sleef_${_component}_FOUND FALSE)
```

Comment:
Not sure what this means: 
> The package configuration file may set <PackageName>_FOUND to false to tell find_package that component requirements are not satisfied.

Do we also need the guard here for optional components, need to check.